[DA] 이미지 분석단계 보여지는 방식 변경

kimkyunghun3 opened the pull request:
# 개요
- 🔗  이슈링크 : #153 

# 변경사항
이미지 분석 단계에서 모달 형식 -> 토스트 뷰 형식으로 변경

## 작업 내용
- indicator 토스트뷰 구현
- OCR 요청 후 응답오는 시간 동안 indicator 토스트뷰 띄우도록 변경
- ViewModel에서 ViewController 역할인 부분 일부 이동
- 타입에 따라 다른 toastView 띄우도록 로직 변경
- 
### 스크린샷


쿠폰 성공| 쿠폰 실패
---|---
![쿠폰성공](https://user-images.githubusercontent.com/52434820/185056896-eecf7f54-57a6-40c1-8603-fbe749881748.gif)|![쿠폰실패](https://user-images.githubusercontent.com/52434820/185057032-45c2a8ef-9845-4830-b107-2692c9270d6a.gif)

kimkyunghun3 committed:
:recycle: 타입에 따라 다른 토스트뷰 보이도록 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: alert 불필요한 코드 제거

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: 쿠폰 이미지 분석동안 토스트뷰 띄우도록 ViewModel 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

kimkyunghun3 committed:
:recycle: 쿠폰 이미지 분석동안 토스트뷰 띄우도록 ViewController 로직 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성

szzang-kr commented on Projects/App/Sources/Main/MainViewController.swift:
```diff
@@ -29,6 +29,14 @@ final class MainViewController: BaseViewController<MainViewModelProtocol> {
     private let toastView = ToastView()
+    private let couponIndicatorToastView = DDIPToastView(.imageCheck)
+
+    lazy var activityIndicator: UIActivityIndicatorView = {
```
MainViewController 외부에서 사용되지 않는다면 private이 필요할것 같습니다!

kimkyunghun3 replied:
아핫 접근제어자 추가했습니다!

szzang-kr commented on Projects/App/Sources/Main/MainViewController.swift:
```diff
@@ -141,3 +174,48 @@ extension MainViewController {
+
+    private func setToastView() {
+        couponIndicatorToastView.setDescriptionLabel("쿠폰 이미지 분석 중이에요!\n 잠시만 기다려주세요!")
+        couponIndicatorToastView.translatesAutoresizingMaskIntoConstraints = false
+        activityIndicator.translatesAutoresizingMaskIntoConstraints = false
```
snapKit을 사용한다면 해당 내용은 필요가 없습니다! 

make 함수 내부에서 설정하기 때문입니다! 😄
SnapKit Library 내부의 `translatesAutoresizingMaskIntoConstraints` 내용을 찾아보시면 도움이 될 것 같습니다.

kimkyunghun3 replied:
내부를 보니까 이렇게 적혀있네요! ConstraintView에 내부에서 설정해주어서 할 필요가없군요! 좋습니다 ㅎㅎ

```swift
extension ConstraintView : LayoutConstraintItem {
}

extension LayoutConstraintItem {
    
    internal func prepare() {
        if let view = self as? ConstraintView {
            view.translatesAutoresizingMaskIntoConstraints = false
        }
    }
}
```

kimkyunghun3 committed:
:recycle: 불필요한 코드 제거 및 접근제어자 변경

# :tada: = 프로젝트 시작 (#)
# :sparkles: = create 파일 생성 시 (#)
# :bulb: = update 코드 및 UI 업데이트, 기능 구현 시 (#)
# :rocket: = release 배포 시 (#)
# :recycle: = refactor 리팩토링 시 (기능은 같은데 코드가 달라지는경우) (#)
# :memo: = 문서 작성 시 (#)
# :art: = 디자인 시스템 관련 내용 (Color / Asset) (#)
# :hammer: = 테스트 코드 작성 (#)
################

# 커밋 타입 : 커밋 제목 (계정명/저장소명#이슈번호) 형식으로 작성하며 제목은 최대 50글자 정도로만 입력
# 제목을 아랫줄에 작성, 제목 끝에 마침표 금지, 무엇을 했는지 명확하게 작성
